Added the Project Structure Guide to docs/guide, and updated README. (#6)
digital-asset/cn-quickstart: README.md

## User Documentation

### User Guides

> There are user guides intended to provide an engineering handover to teams using the Canton Network Quickstart to bootstrap their project.

Reviewer: (nit - pick the version you prefer)
```suggestion
The following user guides provide an engineering handover to teams using the Canton Network Quickstart to bootstrap their project.
```
